lib/puppet/util/windows/service.rb

```diff
@@ -254,15 +254,26 @@ class QUERY_SERVICE_CONFIGW < FFI::Struct
     # @param [:string] service_name name of the service to start
     def start(service_name)
       open_service(service_name, SC_MANAGER_CONNECT, SERVICE_START | SERVICE_QUERY_STATUS) do |service|
-        # don't attempt to fail here if the service isn't stopped because windows error codes
-        # are likely more informative than ours and a failed call to StartServiceW will produce
-        # those errors
-        wait_for_pending_transition(service, SERVICE_STOP_PENDING, SERVICE_STOPPED)
+        # A truthy value return from pending transision indicates the service is now in
+        # SERVICE_RUNNING, a falsey value indicates the service is in another state
+        if wait_on_pending(service, [SERVICE_START_PENDING, SERVICE_CONTINUE_PENDING], SERVICE_RUNNING)
+          # just return if the service is already running
+          return
+        end
+        # ignore the return from the pending transition, we don't care what state the service
```

Comment:
I think these types of comments should maybe go in the method comments? E.g. in `wait_on_pending`, you can have a longer comment describing example uses of the methods. Don't have strong feelings about this though.